Automattic/jetpack
# Connection: do not disconnect sites on WPCOM in Offline Mode

Running `wp jetpack disconnect blog` will disconnect site even if it's currently in Offline Mode.
Besides removing the tokens, it will also send the disconnect API request to WPCOM.

Sites get switched into Offline Mode for a reason, and there's a chance it's a local staging site with tokens copied from the live site. Disconnecting it on WPCOM will break the live site's connection.

D110977-code checks for IDC before disconnecting a site to prevent this, but it requires the `home` and `siteurl` parameters added to the API request. Hovewer, these parameters [don't get added](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/identity-crisis/src/class-identity-crisis.php#L195) in Offline Mode.

So this PR will make sites in Offline Mode skip WPCOM disconnect altogether.

## Proposed changes:
* Do not disconnect sites in Offline Mode on WPCOM.
* Add an mu-plugin for Monorepo to still perform disconnect in CLI, because `DOCKER_REQUEST_URL` trick makes forces Offline Mode over all sites in CLI.

### Other information:


## Jetpack product discussion
Discovered in p1686156171842979-slack-C02NENKA4GN

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Connect Jetpack site
2. Run `wp jetpack disconnect blog` via CLI, confirm that tokens got removed.
3. Check Kibana's `log2logstash` and confirm that the disconnect record is there: `feature:jetpack_xmlrpc_deregister and blog_id:your-blog-id`
4. Connect the site again, switch it to Offline Mode.
5. Run `wp jetpack disconnect blog` via CLI, confirm that tokens got removed.
6. Check Kibana again and confirm that no disconnect records were created.
7. If you tried the above instructions in some other environment, try disconnecting a non-"offline mode" site in Monorepo via same CLI command. Confirm that it got disconnected and the Kibana record got added.
